uboot-mediatek: Add GPT table write command for Arcadyan Mozart

Ansuel opened the pull request:
Add an additional bootmenu option and commands to overwrite the GPT table with the custom one for upstream U-Boot.

This is needed as the pre-installed GPT table differ and updating the GPT table is a requirement to install the upstream U-Boot (or the old U-Boot will be loaded)

To make the migration to new U-Boot bootloader an user can now load the system with mtk_uartboot and select "Load GPT Table ..." in conjunction with other update command.

Ansuel committed:
uboot-mediatek: Add GPT table write command for Arcadyan Mozart

Add an additional bootmenu option and commands to overwrite the GPT
table with the custom one for upstream U-Boot.

This is needed as the pre-installed GPT table differ and updating the
GPT table is a requirement to install the upstream U-Boot (or the old
U-Boot will be loaded)

To make the migration to new U-Boot bootloader an user can now load the
system with mtk_uartboot and select "Load GPT Table ..." in conjunction
with other update command.

Link: https://github.com/openwrt/openwrt/pull/18874
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented on package/boot/uboot-mediatek/patches/456-add-arcadyan-mozart.patch:
```diff
@@ -165,6 +168,7 @@
 +reset_factory=eraseenv && reset
 +emmc_read_production=part start mmc 0 $part_default part_addr && part size mmc 0 $part_default part_size && run mmc_read_vol
 +emmc_read_recovery=part start mmc 0 $part_recovery part_addr && part size mmc 0 $part_recovery part_size && run mmc_read_vol
++emmc_write_gpt=mmc erase 0x0 0x1 && mmc write $fileaddr 0x0 0x1
```
I updated this to actually erase and write for the actual size of a gpt table (34 block)

ynezz commented:
> the bootmenu bug is unrelated to this feature, just triggered it.

Sure, but it makes the bootmenu a bit unusable, because `b. Reset all settings to factory defaults.` gets executed on every UP/DOWN key press, preventing any bootmenu movement using arrow keys.

> For reference and for everyone else, I posted the bootmenu upstream and is waiting review. Finger crossed we have that upstream.

Thanks! For reference its here https://lore.kernel.org/u-boot/20250525134407.3760-1-ansuelsmth@gmail.com/

Ansuel committed:
mediatek: filogic: provide custom GPT table for Arcadyan Mozart

New revision Arcadyan Mozart apply a new partition table and moved the
factory partition to a new location. This conflicts with the standard
partition layout and also make the nvmem cells to not correctly identify
the MAC address anymore.

Generate and Provide a new GPT partition table that account for the new
calibration partition.
The previous revision of the Arcadyan Mozart device is considered
pre-production devices and should not be intended for real support.

Link: https://github.com/openwrt/openwrt/pull/18874
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
I discovered uboot parse numbers as HEX hence with 34 we were erase/writing 0x34... I updated to use HEX.

Also New Mozart need a new GPT table. I added in this commit the required changes to account for that.

ynezz reviewed (approved):
Thanks, works like a charm.

Ansuel commented on target/linux/mediatek/image/filogic.mk:
```diff
@@ -52,6 +52,24 @@ define Build/mt798x-gpt
+			-t 0x83	-N ubootenv	-r	-p 512k@4M \
+			-t 0xef	-N fip		  -r	-p 4M@6656k \
+			-t 0x83	-N factory	-r	-p 8M@25M \
+			-t 0x2e	-N mfgdata	-r	-p 8M@33M \
```
was following original GPT table but I guess they were just wrong with setting this type?

dangowrt replied:
The original table doesn't have `mfgdata`...

Ansuel replied:
You have v1 or v2? It was introduced with v2
